### DISCUSSION TITLE
Prisma types in frontend project

### DISCUSSION THREAD

**Discussion by azertyalex** [prisma/prisma#17771] [Q&A]
### Bug description

According to this release: https://github.com/prisma/prisma/releases/tag/2.13.0

it is possible to import types and enums in the browser, however, when I build my vue project and serve it I get following error

` Error: Dynamic require of "os" is not supported`
Same errors as this issue: https://github.com/prisma/prisma/issues/10058.
Since that issue was closed over a year ago I'm creating a new ticket for this

Is using prisma types in frontend not supported anymore?

### How to reproduce

Use @prisma/client or shared database package(monorepo) in a vue (probably any frontend) framework.

### Prisma information

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "linux-musl"]
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model User {
  name String
}

### Environment & setup

- OS: Ubuntu 22.04
- Database: mysql
- Node.js version: v16.14.2

### Prisma Version

```
prisma                  : 4.8.0
@prisma/client          : 4.9.0
Current platform        : debian-openssl-3.0.x
Query Engine (Node-API) : libquery-engine d6e67a83f971b175a593ccc12e15c4a757f93ffe (at ../../../../../.config/yarn/global/node_modules/@prisma/engines/libquery_engine-debian-openssl-3.0.x.so.node)
Migration Engine        : migration-engine-cli d6e67a83f971b175a593ccc12e15c4a757f93ffe (at ../../../../../.config/yarn/global/node_modules/@prisma/engines/migration-engine-debian-openssl-3.0.x)
Introspection Engine    : introspection-core d6e67a83f971b175a593ccc12e15c4a757f93ffe (at ../../../../../.config/yarn/global/node_modules/@prisma/engines/introspection-engine-debian-openssl-3.0.x)
Format Binary           : prisma-fmt d6e67a83f971b175a593ccc12e15c4a757f93ffe (at ../../../../../.config/yarn/global/node_modules/@prisma/engines/prisma-fmt-debian-openssl-3.0.x)
Format Wasm             : @prisma/prisma-fmt-wasm 4.8.0-61.d6e67a83f971b175a593ccc12e15c4a757f93ffe
Default Engines Hash    : d6e67a83f971b175a593ccc12e15c4a757f93ffe
Studio                  : 0.479.0
```

**Comment by nurul3101** [comment]
Hello @azertyalex 👋 

Importing Prisma types is possible. Are you trying to import the generated `user` type in your frontend application?
Have you configured custom path for generation of PrismaClient or is it in the default location of inside node_modules?
How are you importing the types?

**Reply by azertyalex** [reply]
> Are you trying to import the generated user type in your frontend application?

yes indeed

I have tried both ways you're suggesting and many other ways. I'm using  turborepo and created a database package.

I have imported via @project/database and @prisma/client. Both ways work in development but when serving my dist is where things go wrong

**Comment by azertyalex** [comment]
I seem to have gotten further by using a generated client at a custom path. I've stumbled upon a new error 
```js
Uncaught SyntaxError: The requested module '/@fs/home/<xyz>/Code/<xyz>/j<xyz>/packages/database/src/client/index-browser.js' does not provide an export named 'UserRoleValue' (at auth.store.ts:8:10)
```

Where auth.store.js:8 looks like
```js
import { UserRoleValue } from '@signalserver/database';
```

schema.prisma looks like:
```prisma
enum UserRoleValue {
  SUPERADMIN
  USER
}
```

and the generated type looks like this:
```js

export const UserRoleValue: {
  SUPERADMIN: 'SUPERADMIN',
  USER: 'USER',
};

export type UserRoleValue = (typeof UserRoleValue)[keyof typeof UserRoleValue]

```

**Reply by nurul3101** [reply]
Are the enums that you have defined used in a model? So does a field in any of the models have a reference to `UserRoleValue` enum?

**Reply by mankinchi** [reply]
I'm facing the same issue, with the same error. To answer @nurul3101's question, I have fields that use the enum reference.

**Reply by nurul3101** [reply]
@mankinchi Is your project open source? Any way I can reproduce it locally?

**Reply by mankinchi** [reply]
hi @nurul3101 , this is the minimal application that I can reproduce the issue https://github.com/mankinchi/minimal-prisma-enum

**Reply by nurul3101** [reply]
Thanks for sharing the reproduction. Meanwhile, I would recommend you to check out this repository that users prisma types in frontend (next.js) https://github.com/MarcMogdanz/typed-prisma-package

**Reply by mankinchi** [reply]
Just to be clear, the problem wasn’t with typings, since I can easily import them from prisma generated file. My problem is that I can’t import enum as values to be used in FE

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
